### PR TITLE
`caliper-core`: add tests for `Message` classes

### PR DESCRIPTION
**Pull request by EshaanAgg**
This PR introduces the tests for the base `Message` class in `caliper-core`, and then introduces three similar test suites for the constructors of three types of messages. When approved, I can work on adding similar tests for all the message classes. 

## Checklist
 - [X]  A link to the issue/user story that the pull request relates to
 - [ ]  How to recreate the problem without the fix
 - [ ]  Design of the fix
 - [ ]  How to prove that the fix works
 - [X]  Automated tests that prove the fix keeps on working
 - [ ]  Documentation - any JSDoc, website, or Stackoverflow answers?

## Issue/User story
Fixes part of #1557 

## Steps to Reproduce

1.
2.
3.
4.

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-caliper)
- [ ] [GitHub Issues](https://github.com/hyperledger/caliper/issues)
- [ ] [Discord history](https://discord.com/channels/905194001349627914/941417677778473031)

### CODE REVIEW

**Comment by davidkel** [comment]
@aklenik Can you take a look

**Comment by davidkel** [comment]
On a quick glance, these seem ok to me, but really want @aklenik to take a look as he is more familiar with this part of the code I think

File: packages/caliper-core/test/common/core/messages/message.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            message.hasError().should.be.true;
+        })
+
+        it("should return false if the message has no error", () => {
+            const message = new Message(mockSender, mockRecipients, mockType, mockContent, mockDate);
+            message.hasError().should.be.false;
+        })
+    })
+
+    describe("forRecipient", () => {
```
Comment (davidkel) [inline packages/caliper-core/test/common/core/messages/message.js:93]: We need tests to show that it returns false as well

**Review by davidkel — CHANGES_REQUESTED** [review]
One comment around a couple more tests being required, but these seem reasonable to me.

File: packages/caliper-core/test/common/core/messages/message.js (hunk 1 above)
Comment (EshaanAgg) [reply packages/caliper-core/test/common/core/messages/message.js]: My bad for missing this case! I have updated the test suite for the same. Thanks!

**Review by davidkel — APPROVED** [review]
Looks good to me, many thanks for the contribution

**Pull request merged**